studiopress/pattern-manager
# Allow inserting PM Pattern Blocks if the slug doesn't match the file name

Allow inserting PM Pattern Blocks if the slug doesn't match the filename. 

This fixes the problem with inserting patterns that I mentioned in Sprint Review.

### How to test

1. Activate Twenty Twenty-Three (Twenty Twenty-Three didn't cause this bug, it just revealed it)
2. Create a new pattern
3. Add a 'PM Pattern' block to it
4. Select a Twenty Twenty-Three pattern:

<img width="875" alt="Screenshot 2023-04-24 at 3 15 55 PM" src="https://user-images.githubusercontent.com/4063887/234118813-a4ffcc82-5cdb-436d-842c-d7a31705e552.png">

5. Click 'Create Pattern' to publish it
6. Create a new post (not a pattern)
7. Insert the pattern you created in steps 2-5
8. Expected: The pattern displays: 

<img width="1121" alt="Screenshot 2023-04-24 at 3 17 30 PM" src="https://user-images.githubusercontent.com/4063887/234119134-d40bb07d-8d82-49c4-9067-c82522b68c63.png">

Before, it didn't.

### Documentation
 No documentation required. 



### Notes
In [TwentyTwentythree](https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-content/themes/twentytwentythree/patterns/call-to-action.php#L4) and [Frost](https://github.com/wpengine/frost/blob/6e1fc0527eb91e498a8385f213f991383432d406/patterns/footer-default.php#L4), the slug of the pattern and the file name are different:

<img width="917" alt="Screenshot 2023-04-24 at 2 30 54 PM" src="https://user-images.githubusercontent.com/4063887/234110498-3c0327fa-550f-44dd-95a3-4e362813aa98.png">

That's not wrong, it just points out that my PRs weren't very robust.

My PRs referenced the pattern name, not the slug.

This fixes it to reference the pattern slug.
